src/bootstrap/check.rs

```diff
@@ -115,7 +107,35 @@ pub fn cargo(build: &Build, stage: u32, host: &str) {
     // available.
     cargo.env("CFG_DISABLE_CROSS_TESTS", "1");
 
-    build.run(cargo.env("PATH", newpath));
+    build.run(cargo.env("PATH", &path_for_cargo(build, compiler)));
+}
+
+/// Runs `cargo test` for the rls.
+pub fn rls(build: &Build, stage: u32, host: &str) {
+    let compiler = &Compiler::new(stage, host);
+
+    let mut cargo = build.cargo(compiler, Mode::Tool, host, "test");
+    cargo.arg("--manifest-path").arg(build.src.join("src/tools/rls/Cargo.toml"));
+
+    // Don't build tests dynamically, just a pain to work with
+    cargo.env("RUSTC_NO_PREFER_DYNAMIC", "1");
+
+    build.add_rustc_lib_path(compiler, &mut cargo);
+
+    // Avoid using the rustc shim.
+    cargo.env_remove("RUSTC");
```

Comment:
So, baseline is with the env.remove and setting the PATH, and tests pass. If I remove both of them, I get:

```
thread '<unnamed>' panicked at 'could not run cargo: ChainedError { error: failed to run `rustc` to learn about target-specific information, cause: process didn't exit successfully: `/home/ncameron/rust2/build/bootstrap/debug/rustc - --crate-name ___ --print=file-names -Zunstable-options -Zcontinue-parse-after-error --error-format=json -Zsave-analysis --crate-type bin --crate-type rlib --target x86_64-unknown-linux-gnu` (exit code: 101)
--- stderr
error: Option 'error-format' given more than once.
```

For each test. I think because the rustc shim adds `--error-format` even though the RLS has added it already.